app/assets/src/components/views/amr_heatmap/AMRHeatmapView.jsx

```diff
@@ -66,6 +73,43 @@ export default class AMRHeatmapView extends React.Component {
     });
   };
 
+  updateMaxValue = maxValue => {
+    if (this.state.maxValue !== maxValue) {
+      this.setState({
+        maxValue,
+      });
+    }
+  };
+
+  renderHeader() {
+    return (
+      <ViewHeader className={cs.viewHeader}>
+        <ViewHeader.Content>
+          <ViewHeader.Pretitle>
+            Antimicrobial Resistance Heatmap
+          </ViewHeader.Pretitle>
+          <ViewHeader.Title
+            label={`Comparing ${
+              this.props.sampleIds ? this.props.sampleIds.length : ""
+            } Samples`}
+          />
+        </ViewHeader.Content>
+      </ViewHeader>
+    );
+  }
+
+  renderControls() {
+    return (
+      <AMRHeatmapControls
+        controls={this.assembleControlOptions()}
+        selectedOptions={this.state.selectedOptions}
```

Comment:
We have a convention of extracting `this.props` and `this.state` at the top of the function in the new code we write.

```
  const { selectedOptions } = this.state
```